vms/proposervm/block/build.go

```diff
@@ -55,26 +157,27 @@ func Build(
 			PChainHeight: pChainHeight,
 			Certificate:  cert.Raw,
 			Block:        blockBytes,
+			VRFSig:       blockVrfSig,
 		},
 		timestamp: timestamp,
 		cert:      cert,
 		proposer:  ids.NodeIDFromCert(cert),
 	}
-	var blockIntf SignedBlock = block
 
-	unsignedBytesWithEmptySignature, err := Codec.Marshal(CodecVersion, &blockIntf)
-	if err != nil {
+	var err error
+
+	// temporary, set the bytes to the marshaled content of the block.
+	// this doesn't include the signature ( yet )
+	if block.bytes, err = marshalBlock(block); err != nil {
 		return nil, err
 	}
 
-	// The serialized form of the block is the unsignedBytes followed by the
-	// signature, which is prefixed by a uint32. Because we are marshalling the
-	// block with an empty signature, we only need to strip off the length
-	// prefix to get the unsigned bytes.
-	lenUnsignedBytes := len(unsignedBytesWithEmptySignature) - wrappers.IntLen
-	unsignedBytes := unsignedBytesWithEmptySignature[:lenUnsignedBytes]
-	block.id = hashing.ComputeHash256Array(unsignedBytes)
+	// calculate the block ID.
+	if err = block.initializeID(); err != nil {
+		return nil, err
+	}
```

Comment:
If this is the whole reason why we created this `initializeID` method, I think it would make more sense as a standalone function that is `calculateID` and takes in the `bytes []byte` directly.